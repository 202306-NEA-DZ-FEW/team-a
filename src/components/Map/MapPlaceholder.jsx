function MapPlaceholder() {
  return (
    <div className='h-screen flex gap-10 max-w-5xl mx-auto items-center'>
      <div className='w-[50%] flex flex-col gap-4 items-center'>
        <div className='w-[80%] h-[25px] rounded-full bg-gray-300 animate-pulse' />
        <div className='w-[30%] h-[30px] rounded-full bg-gray-300 animate-pulse' />
        <div className='w-[80%] h-[25px] rounded-full bg-gray-300 animate-pulse' />
        <div className='w-[30%] h-[30px] rounded-full bg-gray-300 animate-pulse' />
        <div className='w-[80%] h-[25px] rounded-full bg-gray-300 animate-pulse' />
        <div className='w-[30%] h-[30px] rounded-full bg-gray-300 animate-pulse' />
      </div>
      <div className='w-[40%] h-[450px] rounded-3xl bg-gray-300 animate-pulse' />
    </div>
  );
}

export default MapPlaceholder;
