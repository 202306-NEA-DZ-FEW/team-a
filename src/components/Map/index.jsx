import Image from "next/image";
import { useTranslation } from "next-i18next";
import CountUp from "react-countup";
function Map() {
  const { i18n, t } = useTranslation();
  return (
    <div className='min-h-screen flex flex-col justify-between items-center max-w-5xl mx-auto mb-10 gap-4 p-4 lg:flex-row-reverse'>
      <figure className='w-[55%]'>
        <Image
          src='/images/dz.svg'
          alt='map illustration'
          width={800}
          height={800}
          className='w-full h-full '
          priority
        />
      </figure>
      <div className='w-[45%] text-center flex flex-col gap-4'>
        <h1 className='text-4xl font-black'>{t("about:growthEstimation")}</h1>
        <div
          className={`flex ${
            i18n.language === "tr" ? "flex-col-reverse" : "flex-col"
          } gap-2`}
        >
          <p className='text-xl md:text-2xl text-accent font-bold'>
            {t("about:bytheEndOf")}
          </p>
          <p className='text-3xl md:text-4xl font-bold'>
            <CountUp duration={7} start={2000} end={t("about:year")} />
          </p>
        </div>
        <div
          className={`flex ${
            i18n.language === "tr" ? "flex-col-reverse" : "flex-col"
          } gap-2`}
        >
          <p className='text-xl md:text-2xl text-accent font-bold'>
            {t("about:helpAim")}
          </p>

          <p className='text-3xl md:text-4xl font-bold'>
            <CountUp duration={7} end={t("about:percentage")} /> %
          </p>
        </div>
        <div
          className={`flex ${
            i18n.language === "tr" ? "flex-col-reverse" : "flex-col"
          } gap-2`}
        >
          <p className='text-xl md:text-2xl text-accent font-bold'>
            {t("about:ofSociety")}
          </p>

          <p
            dir={i18n.language === "ar" ? "rtl" : "ltr"}
            className='text-3xl md:text-4xl font-bold'
          >
            <CountUp duration={7} end={t("about:stateNumber")} />{" "}
            {t("about:states")}
          </p>
        </div>
      </div>
    </div>
  );
}

export default Map;
