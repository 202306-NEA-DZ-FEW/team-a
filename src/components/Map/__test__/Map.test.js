import renderer from "react-test-renderer";

import Map from "..";

it("renders correctly", () => {
  const tree = renderer.create(<Map />).toJSON();
  expect(tree).toMatchSnapshot();
});
