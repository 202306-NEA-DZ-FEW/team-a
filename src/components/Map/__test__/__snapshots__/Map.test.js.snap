// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="min-h-screen flex flex-col justify-between items-center max-w-5xl mx-auto mb-10 gap-4 p-4 lg:flex-row-reverse"
>
  <figure
    className="w-[55%]"
  >
    <img
      alt="map illustration"
      className="w-full h-full "
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height={800}
      onError={[Function]}
      onLoad={[Function]}
      src="/images/dz.svg"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={800}
    />
  </figure>
  <div
    className="w-[45%] text-center flex flex-col gap-4"
  >
    <h1
      className="text-4xl font-black"
    >
      growthEstimation
    </h1>
    <div
      className="flex flex-col gap-2"
    >
      <p
        className="text-xl md:text-2xl text-accent font-bold"
      >
        bytheEndOf
      </p>
      <p
        className="text-3xl md:text-4xl font-bold"
      >
        <span>
          2,000
        </span>
      </p>
    </div>
    <div
      className="flex flex-col gap-2"
    >
      <p
        className="text-xl md:text-2xl text-accent font-bold"
      >
        helpAim
      </p>
      <p
        className="text-3xl md:text-4xl font-bold"
      >
        <span />
         %
      </p>
    </div>
    <div
      className="flex flex-col gap-2"
    >
      <p
        className="text-xl md:text-2xl text-accent font-bold"
      >
        ofSociety
      </p>
      <p
        className="text-3xl md:text-4xl font-bold"
        dir="ltr"
      >
        <span />
         
        states
      </p>
    </div>
  </div>
</div>
`;
